src/modules/authorization/sagas.ts

```diff
@@ -122,6 +133,64 @@ export function createAuthorizationSaga() {
     }
   }
 
+  function* handleFetchAuthorizationRequest(
+    action: FetchAuthorizationRequestAction
+  ) {
+    const { authorization } = action.payload
+
+    try {
+      if (!isValidType(authorization.type)) {
+        throw new Error(`Invalid authorization type ${authorization.type}`)
+      }
+
+      const { chainId } = authorization
+
+      const provider: Provider = yield call(() => getNetworkProvider(chainId))
```

Comment:
Using an arrow function will cause the effect to be untestable, would you mind fixing it?
```suggestion
      const provider: Provider = yield call(getNetworkProvider, chainId)
```